# Lab | Pandas

In this lab, we will be working with the customer data from an insurance company, which can be found in the CSV file located at the following link: https://raw.githubusercontent.com/data-bootcamp-v4/data/main/file1.csv

The data includes information such as customer ID, state, gender, education, income, and other variables that can be used to perform various analyses.

Throughout the lab, we will be using the pandas library in Python to manipulate and analyze the data. Pandas is a powerful library that provides various data manipulation and analysis tools, including the ability to load and manipulate data from a variety of sources, including CSV files.

### Data Description

- Customer - Customer ID

- ST - State where customers live

- Gender - Gender of the customer

- Education - Background education of customers 

- Customer Lifetime Value - Customer lifetime value(CLV) is the total revenue the client will derive from their entire relationship with a customer. In other words, is the predicted or calculated value of a customer over their entire duration as a policyholder with the insurance company. It is an estimation of the net profit that the insurance company expects to generate from a customer throughout their relationship with the company. Customer Lifetime Value takes into account factors such as the duration of the customer's policy, premium payments, claim history, renewal likelihood, and potential additional services or products the customer may purchase. It helps insurers assess the long-term profitability and value associated with retaining a particular customer.

- Income - Customers income

- Monthly Premium Auto - Amount of money the customer pays on a monthly basis as a premium for their auto insurance coverage. It represents the recurring cost that the insured person must pay to maintain their insurance policy and receive coverage for potential damages, accidents, or other covered events related to their vehicle.

- Number of Open Complaints - Number of complaints the customer opened

- Policy Type - There are three type of policies in car insurance (Corporate Auto, Personal Auto, and Special Auto)

- Vehicle Class - Type of vehicle classes that customers have Two-Door Car, Four-Door Car SUV, Luxury SUV, Sports Car, and Luxury Car

- Total Claim Amount - the sum of all claims made by the customer. It represents the total monetary value of all approved claims for incidents such as accidents, theft, vandalism, or other covered events.


External Resources: https://towardsdatascience.com/filtering-data-frames-in-pandas-b570b1f834b9

## Challenge 1: Understanding the data

In this challenge, you will use pandas to explore a given dataset. Your task is to gain a deep understanding of the data by analyzing its characteristics, dimensions, and statistical properties.

- Identify the dimensions of the dataset by determining the number of rows and columns it contains.
- Determine the data types of each column and evaluate whether they are appropriate for the nature of the variable. You should also provide suggestions for fixing any incorrect data types.
- Identify the number of unique values for each column and determine which columns appear to be categorical. You should also describe the unique values of each categorical column and the range of values for numerical columns, and give your insights.
- Compute summary statistics such as mean, median, mode, standard deviation, and quartiles to understand the central tendency and distribution of the data for numerical columns. You should also provide your conclusions based on these summary statistics.
- Compute summary statistics for categorical columns and providing your conclusions based on these statistics.

In [2]:
import pandas as pd
import numpy as np

In [3]:
# loading csv file as df and looking at the first and last 5 rows as well as columns
df = pd.read_csv('https://raw.githubusercontent.com/data-bootcamp-v4/data/main/file1.csv')
display(df)
df.shape

Customer          ST GENDER             Education  \
0     RB50392  Washington    NaN                Master   
1     QZ44356     Arizona      F              Bachelor   
2     AI49188      Nevada      F              Bachelor   
3     WW63253  California      M              Bachelor   
4     GA49547  Washington      M  High School or Below   
...       ...         ...    ...                   ...   
4003      NaN         NaN    NaN                   NaN   
4004      NaN         NaN    NaN                   NaN   
4005      NaN         NaN    NaN                   NaN   
4006      NaN         NaN    NaN                   NaN   
4007      NaN         NaN    NaN                   NaN   

     Customer Lifetime Value   Income  Monthly Premium Auto  \
0                        NaN      0.0                1000.0   
1                 697953.59%      0.0                  94.0   
2                1288743.17%  48767.0                 108.0   
3                 764586.18%      0.0                 106.0   
4                 536307.65%  36357.0                  68.0   
...                      ...      ...                   ...   
4003                     NaN      NaN                   NaN   
4004                     NaN      NaN                   NaN   
4005                     NaN      NaN                   NaN   
4006                     NaN      NaN                   NaN   
4007                     NaN      NaN                   NaN   

     Number of Open Complaints     Policy Type  Vehicle Class  \
0                       1/0/00   Personal Auto  Four-Door Car   
1                       1/0/00   Personal Auto  Four-Door Car   
2                       1/0/00   Personal Auto   Two-Door Car   
3                       1/0/00  Corporate Auto            SUV   
4                       1/0/00   Personal Auto  Four-Door Car   
...                        ...             ...            ...   
4003                       NaN             NaN            NaN   
4004                       NaN             NaN            NaN   
4005                       NaN             NaN            NaN   
4006                       NaN             NaN            NaN   
4007                       NaN             NaN            NaN   

      Total Claim Amount  
0               2.704934  
1            1131.464935  
2             566.472247  
3             529.881344  
4              17.269323  
...                  ...  
4003                 NaN  
4004                 NaN  
4005                 NaN  
4006                 NaN  
4007                 NaN  

[4008 rows x 11 columns]

(4008, 11)

In [4]:
#identifying the amount of rows and columns
print("The data frame has {} rows and {} columns".format(df.shape[0], df.shape[1]))

The data frame has 4008 rows and 11 columns


In [5]:
#Determining the data types of each column. 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4008 entries, 0 to 4007
Data columns (total 11 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Customer                   1071 non-null   object 
 1   ST                         1071 non-null   object 
 2   GENDER                     954 non-null    object 
 3   Education                  1071 non-null   object 
 4   Customer Lifetime Value    1068 non-null   object 
 5   Income                     1071 non-null   float64
 6   Monthly Premium Auto       1071 non-null   float64
 7   Number of Open Complaints  1071 non-null   object 
 8   Policy Type                1071 non-null   object 
 9   Vehicle Class              1071 non-null   object 
 10  Total Claim Amount         1071 non-null   float64
dtypes: float64(3), object(8)
memory usage: 344.6+ KB


In [7]:
# checking if float64 columns are plausible.
df.select_dtypes('float64')

Income  Monthly Premium Auto  Total Claim Amount
0         0.0                1000.0            2.704934
1         0.0                  94.0         1131.464935
2     48767.0                 108.0          566.472247
3         0.0                 106.0          529.881344
4     36357.0                  68.0           17.269323
...       ...                   ...                 ...
4003      NaN                   NaN                 NaN
4004      NaN                   NaN                 NaN
4005      NaN                   NaN                 NaN
4006      NaN                   NaN                 NaN
4007      NaN                   NaN                 NaN

[4008 rows x 3 columns]

In [8]:
# checking if object columns are plausible.
df.select_dtypes('object')

Customer          ST GENDER             Education  \
0     RB50392  Washington    NaN                Master   
1     QZ44356     Arizona      F              Bachelor   
2     AI49188      Nevada      F              Bachelor   
3     WW63253  California      M              Bachelor   
4     GA49547  Washington      M  High School or Below   
...       ...         ...    ...                   ...   
4003      NaN         NaN    NaN                   NaN   
4004      NaN         NaN    NaN                   NaN   
4005      NaN         NaN    NaN                   NaN   
4006      NaN         NaN    NaN                   NaN   
4007      NaN         NaN    NaN                   NaN   

     Customer Lifetime Value Number of Open Complaints     Policy Type  \
0                        NaN                    1/0/00   Personal Auto   
1                 697953.59%                    1/0/00   Personal Auto   
2                1288743.17%                    1/0/00   Personal Auto   
3                 764586.18%                    1/0/00  Corporate Auto   
4                 536307.65%                    1/0/00   Personal Auto   
...                      ...                       ...             ...   
4003                     NaN                       NaN             NaN   
4004                     NaN                       NaN             NaN   
4005                     NaN                       NaN             NaN   
4006                     NaN                       NaN             NaN   
4007                     NaN                       NaN             NaN   

      Vehicle Class  
0     Four-Door Car  
1     Four-Door Car  
2      Two-Door Car  
3               SUV  
4     Four-Door Car  
...             ...  
4003            NaN  
4004            NaN  
4005            NaN  
4006            NaN  
4007            NaN  

[4008 rows x 8 columns]


#### I have found a few suggestions for the data cleaning process:

1. "Customer Lifetime Value" is showing entries with a percentage. The data type object is hence not correct.
I would suggest changing it to a float64 . As I'm not sure how the "%" is behaving when using .astype, I would make a copy first.
In case of errors, I would replace the "%" first.

2. "Number of Open Complaints" has a strange format showing 1/00/0.
Converting 1/0/00 to 1 might be a solution but needs further checking. Afterwards I would performing .astype int64. 

3. In general, most column names have an empty space that I will fill with "_"

In [9]:
#Identify the number of unique values for each column and determine which columns appear to be categorical. 
df.nunique()

Customer                     1071
ST                              8
GENDER                          5
Education                       6
Customer Lifetime Value      1027
Income                        774
Monthly Premium Auto          132
Number of Open Complaints       6
Policy Type                     3
Vehicle Class                   6
Total Claim Amount            761
dtype: int64

In [33]:
categ_col = ["ST", "GENDER", "Education", "Policy Type", "Vehicle Class"]
num_col = ["Income", "Monthly Premium Auto", "Total Claim Amount"]

In [30]:
#Describing the unique values of each categorical column:

In [31]:
for i in categ_col:
    print(f"The unique values in column {i} are:")
    print(df[i].unique())
    print("")

The unique values in column ST are:
['Washington' 'Arizona' 'Nevada' 'California' 'Oregon' 'Cali' 'AZ' 'WA'
 nan]

The unique values in column GENDER are:
[nan 'F' 'M' 'Femal' 'Male' 'female']

The unique values in column Education are:
['Master' 'Bachelor' 'High School or Below' 'College' 'Bachelors' 'Doctor'
 nan]

The unique values in column Policy Type are:
['Personal Auto' 'Corporate Auto' 'Special Auto' nan]

The unique values in column Vehicle Class are:
['Four-Door Car' 'Two-Door Car' 'SUV' 'Luxury SUV' 'Sports Car'
 'Luxury Car' nan]



#### Insights:

ST shows names of states. However the writing style differes. While entries display a full name, some show the abbreviation like 'AZ' and 'WA'. This needs harmonization

GENDER needs harmonization. Either F/M, Female/Male, or female/male.

Bachelor and Bachelors need to be unified.

Policy Type shows 3 distinct values as well as 'nan'. This seems fine.

Vehicle Class SUV is equal to  Luxury SUV and needs to be changed.

In [34]:
# Range of values for numerical columns

for i in num_col:
    min_val = df[i].min()
    max_val = df[i].max()
    print(f"Column {i}'s smallest value is {min_val} and the highest value is {max_val}.")


Column Income's smallest value is 0.0 and the highest value is 99960.0.
Column Monthly Premium Auto's smallest value is 61.0 and the highest value is 35354.0.
Column Total Claim Amount's smallest value is 0.382107 and the highest value is 2893.239678.


In [38]:
#Compute summary statistics such as mean, median, mode, standard deviation, and quartiles to understand the central tendency and distribution of the data for numerical columns. 
#You should also provide your conclusions based on these summary statistics.
df.describe()

Income  Monthly Premium Auto  Total Claim Amount
count   1071.000000           1071.000000         1071.000000
mean   39295.701214            193.234360          404.986909
std    30469.427060           1601.190369          293.027260
min        0.000000             61.000000            0.382107
25%    14072.000000             68.000000          202.157702
50%    36234.000000             83.000000          354.729129
75%    64631.000000            109.500000          532.800000
max    99960.000000          35354.000000         2893.239678

In [50]:
# Compute summary statistics for categorical columns and providing your conclusions based on these statistics.

print(f"The elements of the categorical columns show the following distribution:")
for i in categ_col:
    print(df[i].value_counts())
    print("")

The elements of the categorical columns show the following distribution:
ST
Oregon        320
California    211
Arizona       186
Cali          120
Nevada         98
Washington     81
WA             30
AZ             25
Name: count, dtype: int64

GENDER
F         457
M         413
Male       39
female     28
Femal      17
Name: count, dtype: int64

Education
Bachelor                324
College                 313
High School or Below    296
Master                   94
Doctor                   37
Bachelors                 7
Name: count, dtype: int64

Policy Type
Personal Auto     780
Corporate Auto    234
Special Auto       57
Name: count, dtype: int64

Vehicle Class
Four-Door Car    576
Two-Door Car     205
SUV              199
Sports Car        57
Luxury SUV        20
Luxury Car        14
Name: count, dtype: int64



#### Insights:

State Column shows that most customers live in Oregon, followed by California, Arizona, Cali, Nevada, Washington. WA and AZ only amount for a small amount -> will be merged with Washington & Arizona respectively.

Most Gender counts are within the elements F and M. Therefore, "Male" will be substituted with "M". "female", "Femal" will be substituted with "F".

Most customers have a Bachelor degree. Only a few have doctor. "Bachelors" entries will be counted as "Bachelor".

Most customers have a personal car, followed by company car, and last a special car.

Most customers have a Four-Door Cars, only a few Luxury SUVs or Luxury Cars.

## Challenge 2: analyzing the data

### Exercise 1

The marketing team wants to know the top 5 less common customer locations. Create a pandas Series object that contains the customer locations and their frequencies, and then retrieve the top 5 less common locations in ascending order.

In [74]:
# First, get a general overview of all locations:
df['ST'].value_counts()

ST
Oregon        320
California    211
Arizona       186
Cali          120
Nevada         98
Washington     81
WA             30
AZ             25
Name: count, dtype: int64

In [75]:
# Then, only look at the top 5 less common locations are AZ, WA, Washington, Nevada, and Cali.
df['ST'].value_counts(ascending=True).iloc[0:5]

ST
AZ             25
WA             30
Washington     81
Nevada         98
Cali          120
Name: count, dtype: int64

### Exercise 2

The sales team wants to know the total number of policies sold for each type of policy. Create a pandas Series object that contains the policy types and their total number of policies sold, and then retrieve the policy type with the highest number of policies sold.

*Hint:*
- *Using value_counts() method simplifies this analysis.*
- *Futhermore, there is a method that returns the index of the maximum value in a column or row.*


In [82]:
# Listing up the total number of policies sold per policy type:
df['Policy Type'].value_counts()

Policy Type
Personal Auto     780
Corporate Auto    234
Special Auto       57
Name: count, dtype: int64

In [111]:
# Policy Type in which the highest amount of policies are sold.
df['Policy Type'].value_counts(ascending=True).iloc[0:1]

Policy Type
Special Auto    57
Name: count, dtype: int64

### Exercise 3

The sales team wants to know if customers with Personal Auto have a lower income than those with Corporate Auto. How does the average income compare between the two policy types?

- Use *loc* to create two dataframes: one containing only Personal Auto policies and one containing only Corporate Auto policies.
- Calculate the average income for each policy.
- Print the results.

In [150]:
df_personal_auto = pd.DataFrame(df[ df['Policy Type'] == "Personal Auto" ]).loc[:,['Policy Type', 'Income']]
df_corporate_auto = pd.DataFrame(df[ df['Policy Type'] == "Corporate Auto" ]).loc[:,['Policy Type', 'Income']]

In [161]:
print("The average income of the customers owning a personal car is:")
print(df_personal_auto['Income'].mean())
print("The average income of the customers owning a corporate car is:")
print(df_corporate_auto['Income'].mean())

The average income of the customers owning a personal car is:
38180.69871794872
The average income of the customers owning a corporate car is:
41390.31196581197


In [163]:
income_gap = df_corporate_auto['Income'].mean() - df_personal_auto['Income'].mean()
print(f"The average income of the customers owning a personal car is {income_gap} lower than of those owning a corporate car.")

The average income of the customers owning a personal car is 3209.61324786325 lower than of those owning a corporate car.


### Bonus: Exercise 4


Your goal is to identify customers with a high policy claim amount.

Instructions:

- Review again the statistics for total claim amount to gain an understanding of the data.
- To identify potential areas for improving customer retention and profitability, we want to focus on customers with a high policy claim amount. Consider customers with a high policy claim amount to be those in the top 25% of the total claim amount. Create a pandas DataFrame object that contains information about customers with a policy claim amount greater than the 75th percentile.
- Use DataFrame methods to calculate summary statistics about the high policy claim amount data. 

*Note: When analyzing data, we often want to focus on certain groups of values to gain insights. Percentiles are a useful tool to help us define these groups. A percentile is a measure that tells us what percentage of values in a dataset are below a certain value. For example, the 75th percentile represents the value below which 75% of the data falls. Similarly, the 25th percentile represents the value below which 25% of the data falls. When we talk about the top 25%, we are referring to the values that fall above the 75th percentile, which represent the top quarter of the data. On the other hand, when we talk about the bottom 25%, we are referring to the values that fall below the 25th percentile, which represent the bottom quarter of the data. By focusing on these groups, we can identify patterns and trends that may be useful for making decisions and taking action.*

*Hint: look for a method that gives you the percentile or quantile 0.75 and 0.25 for a Pandas Series.*

*Hint 2: check `Boolean selection according to the values of a single column` in https://towardsdatascience.com/filtering-data-frames-in-pandas-b570b1f834b9*

In [190]:
# Generating an overview of Total Claim Amount Statistics and saving them in a series as claim_amount_stats.
claim_amount_stats = df['Total Claim Amount'].describe()
claim_amount_stats

count    1071.000000
mean      404.986909
std       293.027260
min         0.382107
25%       202.157702
50%       354.729129
75%       532.800000
max      2893.239678
Name: Total Claim Amount, dtype: float64

In [201]:
# The indices 4 and 6 access the 25%- and 75%- percentile of the claim_amount_stats series.
# I saved them in respective variables to access them later.
# Only percentile 75 is needed, as it represents the top 25% customers with the highest claim amount.
percentile_25 = claim_amount_stats[4]
percentile_75 = claim_amount_stats[6]

In [205]:
# The data frame shows  the customers with a high policy claim amount - those in the top 25% of the total claim amount.
# In total, 273 customers belong to that group.
df_customers_percentile_75 = pd.DataFrame(df[ df['Total Claim Amount'] >= percentile_75])
df_customers_percentile_75

Customer          ST GENDER Education Customer Lifetime Value   Income  \
1     QZ44356     Arizona      F  Bachelor              697953.59%      0.0   
2     AI49188      Nevada      F  Bachelor             1288743.17%  48767.0   
17    OE15005        Cali    NaN   College              394524.16%  28855.0   
23    TZ98966      Nevada    NaN  Bachelor              245019.10%      0.0   
26    US89481  California    NaN  Bachelor              394637.21%      0.0   
...       ...         ...    ...       ...                     ...      ...   
1059  YG44474      Oregon      M   College             1401472.13%  54193.0   
1061  RY92647        Cali      F  Bachelor             1050677.17%      0.0   
1068  GS98873     Arizona      F  Bachelor              323912.47%  16061.0   
1069  CW49887  California      F    Master              462680.11%  79487.0   
1070  MY31220  California      F   College              899704.02%  54230.0   

      Monthly Premium Auto Number of Open Complaints     Policy Type  \
1                     94.0                    1/0/00   Personal Auto   
2                    108.0                    1/0/00   Personal Auto   
17                   101.0                    1/0/00   Personal Auto   
23                    73.0                    1/3/00  Corporate Auto   
26                   111.0                    1/0/00   Personal Auto   
...                    ...                       ...             ...   
1059                 117.0                    1/0/00  Corporate Auto   
1061                  92.0                    1/0/00   Personal Auto   
1068                  88.0                    1/0/00   Personal Auto   
1069                 114.0                    1/0/00    Special Auto   
1070                 112.0                    1/0/00   Personal Auto   

      Vehicle Class  Total Claim Amount  
1     Four-Door Car         1131.464935  
2      Two-Door Car          566.472247  
17              SUV          647.442031  
23    Four-Door Car          554.376763  
26    Four-Door Car          799.200000  
...             ...                 ...  
1059            SUV          720.752945  
1061  Four-Door Car          546.524896  
1068  Four-Door Car          633.600000  
1069            SUV          547.200000  
1070   Two-Door Car          537.600000  

[273 rows x 11 columns]

In [222]:
# In the following a few standard statistics about that group are listed.
df_customers_percentile_75.describe()

Income  Monthly Premium Auto  Total Claim Amount
count    273.000000            273.000000          273.000000
mean   24106.545788            163.000000          774.005353
std    27542.717560            613.645376          291.304731
min        0.000000             63.000000          532.800000
25%        0.000000             99.000000          595.200000
50%    18846.000000            113.000000          673.342650
75%    44897.000000            133.000000          844.481918
max    99316.000000          10202.000000         2893.239678

In [225]:
# the median of Total Claim Amount
df_customers_percentile_75['Total Claim Amount'].median()

673.34265

### Insights

The average income of the group is 24,107, with the highest income being 99,316 and lowest income 0. This is resulting in a high standard deviation.

The monthly premium amount a customer pays for their auto insurance ranges between 63 and 10,202.

The average claim amount is 774, with the highest claim being 2893 and the lowest 533. I would assume that there are only a few higher outliers as the mean is not too far away from the median (673).

In [212]:
print(f"The elements of the categorical columns show the following distribution:")
for i in categ_col:
    print(df_customers_percentile_75[i].value_counts())
    print("")

The elements of the categorical columns show the following distribution:
ST
Oregon        93
California    67
Arizona       42
Nevada        29
Cali          21
AZ            11
WA            10
Name: count, dtype: int64

GENDER
F         118
M         103
Male       17
female      5
Femal       2
Name: count, dtype: int64

Education
High School or Below    90
Bachelor                86
College                 84
Master                   9
Doctor                   4
Name: count, dtype: int64

Policy Type
Personal Auto     195
Corporate Auto     65
Special Auto       13
Name: count, dtype: int64

Vehicle Class
SUV              105
Four-Door Car     74
Two-Door Car      33
Sports Car        31
Luxury SUV        19
Luxury Car        11
Name: count, dtype: int64



State Column shows that most customers live in Oregon, followed by California, Arizona, Nevada, Cali, ... this is almost similar to the total group of customers.

Most Gender counts are within the elements F and M. The distribution is similar to the total customer group.

Most customers have a high school degree or lower. In the total group there were much more with a bachelor.

Most customers own a personal car, followed by a corporate car, and last a special car. Same as in the total group.

The group's customers own mostly SUV vehicles (which need to be added to Luxury SUVs), followed by Four-Door cars, which were the most represented in the overall group. 

In the following, some more calculations I made. They're not part of my solution but experimented a bit in the meantime and didn't want to delete them

In [226]:
# These are the customers with a low policy claim amount to be those in the lower 25% of the total claim amount.
# In total, 268 customers belong to that group.
df_customers_percentile_25 = pd.DataFrame(df[ df['Total Claim Amount'] <= percentile_25])
df_customers_percentile_25

Customer          ST GENDER             Education  \
0     RB50392  Washington    NaN                Master   
4     GA49547  Washington      M  High School or Below   
5     OC83172      Oregon      F              Bachelor   
15    BW63560      Oregon    NaN              Bachelor   
21    QK46697  Washington      M             Bachelors   
...       ...         ...    ...                   ...   
1050  BJ53923     Arizona      M  High School or Below   
1051  CZ96653      Oregon      F              Bachelor   
1056  PI47776      Oregon      F              Bachelor   
1063  GQ66762        Cali      M               College   
1065  XD85577        Cali      M              Bachelor   

     Customer Lifetime Value   Income  Monthly Premium Auto  \
0                        NaN      0.0                1000.0   
4                 536307.65%  36357.0                  68.0   
5                 825629.78%  62902.0                  69.0   
15                746313.94%  24599.0                  64.0   
21                617710.93%  61040.0                  79.0   
...                      ...      ...                   ...   
1050              260027.21%  51978.0                  66.0   
1051              853479.28%  47325.0                 107.0   
1056              253070.51%  89451.0                  63.0   
1063              477368.64%  33701.0                  63.0   
1065              284226.69%  69417.0                  73.0   

     Number of Open Complaints     Policy Type  Vehicle Class  \
0                       1/0/00   Personal Auto  Four-Door Car   
4                       1/0/00   Personal Auto  Four-Door Car   
5                       1/0/00   Personal Auto   Two-Door Car   
15                      1/1/00  Corporate Auto  Four-Door Car   
21                      1/1/00   Personal Auto   Two-Door Car   
...                        ...             ...            ...   
1050                    1/0/00  Corporate Auto  Four-Door Car   
1051                    1/0/00   Personal Auto            SUV   
1056                    1/0/00  Corporate Auto  Four-Door Car   
1063                    1/0/00   Personal Auto  Four-Door Car   
1065                    1/1/00   Personal Auto  Four-Door Car   

      Total Claim Amount  
0               2.704934  
4              17.269323  
5             159.383042  
15             42.920271  
21             20.382876  
...                  ...  
1050          144.782152  
1051           64.598216  
1056           61.769564  
1063          171.325856  
1065           30.874869  

[268 rows x 11 columns]

In [213]:
df[ df['Total Claim Amount'] >= percentile_75].shape

(273, 11)

In [214]:
# Total Claim Amount
sum_total_claim = df['Total Claim Amount'].sum()

In [215]:
# Defining Max Claim Amount
max_claim = df['Total Claim Amount'].max()
print(max_claim)

2893.239678


In [216]:
# This is the customer with the highest claim amount.
df[ df['Total Claim Amount'] == max_claim]

Customer       ST GENDER Education Customer Lifetime Value   Income  \
888  BZ65376  Arizona      M  Bachelor              858127.87%  27689.0   

     Monthly Premium Auto Number of Open Complaints    Policy Type  \
888                 239.0                    1/2/00  Personal Auto   

    Vehicle Class  Total Claim Amount  
888    Luxury SUV         2893.239678

In [217]:
# Percentage of total - calculation
perc_of_total = max_claim / sum_total_claim
print(perc_of_total)

0.006670431920344303
